src/ctap/storage.rs

```diff
@@ -570,6 +570,38 @@ impl PersistentStore {
         Ok(self.store.insert(key::AAGUID, aaguid)?)
     }
 
+    /// Returns the upgrade public key if defined for the identifier.
+    pub fn _upgrade_public_key(
```

Comment:
IIRC a function with an underscore prefix is equivalent to implicitly adding the `allow(dead_code)` attribute. Is that intended here?

Reply:
Yes, see [this discussion](https://github.com/google/OpenSK/pull/359#discussion_r682787364) for the rationale. Now that I know it's recursive, I think the best approach is to use `warn(dead_code)`. This should only be allowed in `develop` branch. And the warning during compilation reminds that there is some work in progress. And if the work ends up being deprecated, then we know we can delete the function.